sdk/core/core/inc/internal/az_span_def.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_SPAN_DEF_H
+#define AZ_SPAN_DEF_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * An immutable span of bytes (octets).
+ */
+typedef struct {
```

Comment:
This struct and perhaps some macros/functions that operate on it (if customers need them), need to be public. So, I'm surprised to see this .h file in an internal subdir.

Reply:
yeah, this struct is available to customers through az_span (which includes this definition)
The reason this struct is moved to internal is because we want customers to use az_span.h to get this struct and functionality instead of allowing including struct only

Reply:
Right, but the function definitions were together with the struct before the move?